src/citrine/resources/dataset.py

```diff
@@ -231,9 +231,14 @@ def register_all(self, data_concepts_resources: List[ResourceType]) -> List[Reso
 
         :return the registered versions
         """
-        return [self.register(resource) for resource
```

Comment:
Is this change necessary?  I would have thought that `self.register(resource)` would return the resource with the uid set.

Reply:
More specifically, the issue occurs when trying to store an object that has a reference to an object that doesn't have a uid